File: scrapers/add_programs_to_db.js
import fs from 'fs';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const programsPath = join(__dirname, 'data', 'programs_items.json');
const dbPath = join(__dirname, 'db.json');

const programsData = JSON.parse(fs.readFileSync(programsPath, 'utf8'));
const dbData = JSON.parse(fs.readFileSync(dbPath, 'utf8'));

const programsWithIds = programsData.map((program, index) => {
  return {
    id: index + 1,
    title: program.title,
    school_label: program.school_label,
    level: program.level,
    link: program.link
  };
});

dbData.programs = programsWithIds;

fs.writeFileSync(dbPath, JSON.stringify(dbData, null, 2), 'utf8');

console.log(`✅ Добавлено ${programsWithIds.length} программ в db.json`);
console.log(`📊 Всего программ: ${programsWithIds.length}`);

File: scrapers/add_programs_to_db.py
import json
import os

script_dir = os.path.dirname(os.path.abspath(__file__))

programs_path = os.path.join(script_dir, 'data', 'programs_items.json')
db_path = os.path.join(script_dir, 'db.json')

print(f'Читаю программы из: {programs_path}')
with open(programs_path, 'r', encoding='utf-8') as f:
    programs_data = json.load(f)

print(f'Читаю db.json из: {db_path}')
with open(db_path, 'r', encoding='utf-8') as f:
    db_data = json.load(f)

print('Обрабатываю программы...')
programs_with_ids = []
for index, program in enumerate(programs_data, start=1):
    programs_with_ids.append({
        'id': index,
        'title': program.get('title', ''),
        'school_label': program.get('school_label', ''),
        'level': program.get('level', ''),
        'link': program.get('link', '')
    })

db_data['programs'] = programs_with_ids

print(f'Сохраняю {len(programs_with_ids)} программ в db.json...')
with open(db_path, 'w', encoding='utf-8') as f:
    json.dump(db_data, f, ensure_ascii=False, indent=2)

print(f'✅ Успешно добавлено {len(programs_with_ids)} программ в db.json')
print(f'📊 Всего программ в базе: {len(programs_with_ids)}')

